src/plugins/data_source_management/public/components/direct_query_data_sources_components/associated_object_management/utils/associated_objects_tab_utils.tsx

```diff
@@ -43,49 +43,38 @@ export const isCatalogCacheFetching = (...statuses: DirectQueryLoadingStatus[])
   );
 };
 
-export const redirectToExplorerWithDataSrc = (
+export const redirectToDiscoverWithDataSrc = (
   datasourceName: string,
-  datasourceType: string,
+  datasourceMDSId: string | undefined,
   databaseName: string,
   tableName: string,
   application: ApplicationStart
 ) => {
-  const queryIndex = `${datasourceName}.${databaseName}.${tableName}`;
-  redirectToExplorerWithQuery(datasourceName, datasourceType, queryIndex, application);
-};
-
-export const redirectToExplorerOSIdx = (indexName: string, application: ApplicationStart) => {
-  redirectToExplorerWithQuery(
-    DEFAULT_DATA_SOURCE_NAME,
-    DEFAULT_DATA_SOURCE_TYPE,
-    indexName,
-    application
-  );
-};
-
-export const redirectToExplorerS3 = (datasourceName: string, application: ApplicationStart) => {
-  application.navigateToApp(observabilityLogsID, {
-    path: `#/explorer`,
-    state: {
-      datasourceName,
-      datasourceType: DATA_SOURCE_TYPES.S3Glue,
-    },
+  application.navigateToApp('data-explorer', {
+    path: `discover#?_a=(discover:(columns:!(_source),isDirty:!f,sort:!()),metadata:(view:discover))&_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-15m,to:now))&_q=(filters:!(),query:(dataset:(dataSource:(id:'${
```

Comment:
nit: Feel like these two functions have a lot of duplicate code, could be worth a try to either combine them or wrap the actual path in another function.

Reply:
Originally when it was just the 'state' being redirected to explorer through navigateToApp, that made a lot of sense. However, since Discover's url redirection has different forms between redirecting to a flint index vs an index pattern, the differences i would need to put in between both strings would be too much for it to be worth it, as I think it would greatly reduce readability.